week3/RO_Flowsheet_ERD_Techno_Economics_Analysis.ipynb

```diff
@@ -0,0 +1,1492 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a2a953db",
+   "metadata": {},
+   "source": [
+    "# WaterTAP Academy Week 3 Hands-On Tutorial  \n",
+    "## Techno-Economic Analysis of an RO Flowsheet with an ERD\n",
+    "\n",
+    "In this week’s hands-on tutorial, we will use WaterTAP to perform a Techno-Economic Analysis (TEA) on the flowsheet shown below: a single-stage Reverse Osmosis (RO) system equipped with an Energy Recovery Device (ERD). This example builds directly on the Week 2 after-session assignment.\n",
+    "\n",
+    "<center><img src=\"./graphics/RO_with_ERD.png\" width=\"600\" /></center>\n",
+    "\n",
+    "In the previous tutorial, we introduced how to add costing blocks on top of individual unit models for TEA. Now we will extend those concepts to the **flowsheet level**, demonstrating how performance modeling, costing, and optimization come together for system-wide analysis.\n",
+    "\n",
+    "---\n",
+    "\n",
+    "## In this tutorial, we will work through the following parts:\n",
+    "\n",
+    "**Part 1 – Develop Process Model**  \n",
+    "- Build the RO flowsheet and set up a full process simulation for performance prediction.\n",
+    "\n",
+    "**Part 2 – Develop TEA Model**  \n",
+    "- Integrate WaterTAP’s costing framework to complete the TEA model.  \n",
+    "- Compute key economic and energy metrics, such as Levelized Cost of Water (LCOW) \n",
+    "- Plot and interpret system performance and cost results.\n",
+    "\n",
+    "**Part 3 – Optimization Study for Cost Minimization**  \n",
+    "- Modify the model to set up and solve an optimization problem aiming to minimize LCOW.  \n",
+    "- Review the resulting optimal operating conditions.\n",
+    "\n",
+    "**Part 4 – Try It Yourself: Build Your Optimization and Visualize Results**  \n",
+    "- Create additional visualizations or summary plots for the optimization study.\n",
+    "\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1b96bd83",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "## **Part 1 – Develop Process Model** \n",
+    "\n",
+    "In this part, we will build a predictive process model of the RO flowsheet with ERD.\n",
+    "All steps and code used here were covered in the previous session, so this section serves as a quick review of how to construct a working flowsheet simulation before moving on to TEA and optimization.\n",
+    "\n",
+    "1. **Import Required Modules**  \n",
+    "2. **Create the Process Model**  \n",
+    "3. **Solve the Model**  \n",
+    "4. **Check the Model Prediction**\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b8315593",
+   "metadata": {},
+   "source": [
+    "### **Import Required Modules**\n",
+    "Import all WaterTAP, Pyomo, and helper modules needed for building and solving the RO flowsheet.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
```

Comment:
Do we want this tutorial to be pre-run (having all the cells ran when the user opens the file)? Also, this is coming up as the 2nd cell being run when it should be the first. Perhaps you accidentally ran this cell twice?